django_ipam/tests/base/test_models.py

```diff
@@ -142,6 +142,26 @@ def test_invalid_master_subnet(self):
         if failed:
             self.fail('ValidationError not raised')
 
+    def test_correct_master_subnet_pass(self):
+        subnet1 = self._create_subnet(subnet='12.0.56.0/24')
+        try:
+            subnet2 = self._create_subnet(subnet='12.0.56.0/25', master_subnet=subnet1)
+            self._create_subnet(subnet='12.0.56.0/26', master_subnet=subnet2)
+        except ValidationError:
+            self.fail('Correct master_subnet not accepted')
+
+    def test_correct_master_subnet_fail(self):
```

Comment:
I don't fully understand  the naming `test_correct_master_subnet_fail`, what is this test checking?

Reply:
`test_correct_master_subnet_pass` & `test_correct_master_subnet_fail` are testing the fact that any grandparent subnet should not be taken as a master subnet while parent subnets should be allowed.

`test_correct_master_subnet_pass` makes a `/24` subnet, then `/25` with master as `/24` and `/26` with master `/25`, which should pass
`test_correct_master_subnet_fail` makes a `/24` subnet, then `/25` with master as `/24` and `/26` with master `/24`, which should fail because `/24` is a grandparent of `/26` since `/25` exists in between the two.

I'll think of a more intuative name for these functions and get back. :smile:

Reply:
renamed to `test_invalid_subnet_relation_tree`.

Reply:
Great, thanks